services/ptrarchive.go

```diff
@@ -58,14 +62,45 @@ func (p *PTRArchive) OnDNSRequest(ctx context.Context, req *requests.DNSRequest)
 	p.CheckRateLimit()
 	bus.Publish(requests.SetActiveTopic, p.String())
 
+	v := url.Values{}
+	v.Set("name", "Ava")
+
+	dial := net.Dialer{}
+	client := &outhttp.Client{
+		Transport: &outhttp.Transport{
+			DialContext:         dial.DialContext,
+			TLSHandshakeTimeout: 10 * time.Second,
+		},
+	}
+
 	url := p.getURL(req.Domain)
-	page, err := http.RequestWebPage(url, nil, nil, "", "")
+
+	request, err := outhttp.NewRequest("GET", url, strings.NewReader(v.Encode()))
+
+	cookie := &outhttp.Cookie{
+		Name:   "test",
+		Domain: "ptrarchive.com",
+		Value:  "123432",
+	}
+	request.AddCookie(cookie)
```

Comment:
Creating a whole new `http.Client` just to add a cookie is a bit excessive. I think you could work around this by manually setting a Cookie header through the `hvals` parameter to `RequestWebPage()`

Reply:
That would add a cookie to every requests, is it fine ? Otherwise, I can an optional argument in the method signature to precise whether the request should contain a cookie.

Reply:
I'm thinking something like this: 
```go
cookieHeader := map[string]string{"Cookie", "test=123432"}
page, err := http.RequestWebPage(url, nil, cookieHeader, "", "")
```
This should only add the cookie to the single request.